Source/WTF/wtf/HashTable.h

```diff
@@ -988,18 +978,18 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(HashTable);
                 else if (HashTranslator::equal(Extractor::extract(*entry), key))
                     return AddResult(makeKnownGoodIterator(entry), false);
             }
-#if DUMP_HASHTABLE_STATS
+
             ++probeCount;
+
+#if DUMP_HASHTABLE_STATS
             HashTableStats::recordCollisionAtCount(probeCount);
 #endif
 
 #if DUMP_HASHTABLE_STATS_PER_TABLE
             m_stats->recordCollisionAtCount(probeCount);
 #endif
 
-            if (k == 0)
-                k = 1 | doubleHash(h);
-            i = (i + k) & sizeMask;
+            i = (i + probeCount) & sizeMask;
```

Comment:
I'm not super familiar with quadratic probing, but based on skimming Wikipedia, it seems like the standard formulation would be:

i = (h + probeCount * probeCount) & sizeMask

Is this equivalent?

Reply:
Adding value can be arbitrary quadratic polynomial. Wikipedia is using `probeCount * probeCount` (https://en.wikipedia.org/wiki/Quadratic_probing), but anything is fine so long as it is quadratic polynomial.
In this case, since probeCount is incremented continuously, then,

`h(k, i) = h(k) + (i * (i + 1) / 2)`

In wikipedia case, it is,

`h(k, i) = h(k) + i * i`

Adding `probeCount` is cheap, but we can easily achieve quadratic polynomial.